libraries/AP_HAL/Util.cpp

```diff
@@ -60,6 +60,7 @@ int AP_HAL::Util::vsnprintf(char* str, size_t size, const char *format, va_list
     // null terminate
     int ret = buf._offs;
     buf.write(0);
+    str[size - 1] = '\0'; // really terminate incase the buf.write refused
```

Comment:
```suggestion
    str[ret] = '\0';
```

And remove the `buf.write(0);`